### PR TITLE
updated vis

### PR DESCRIPTION
**Pull request by Kelvinrr**

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -2,8 +2,11 @@ name: knoten
 channels:
   - conda-forge
   - usgs-astrogeology
+  - plotly
 dependencies:
-  - bokeh
+  - plotly
```
Comment (kaitlyndlee) [inline environment.yml]: Looks like the rest of the file is in abc order. Could you move the imports to follow the pattern?

**Comment by kaitlyndlee** [comment]
Will this be the final version of the ctx notebook? It looks like creating the isd got added into the notebook. Do we want this in all the notebooks now?

We also need to clean up the ale imports.

**Comment by Kelvinrr** [comment]
I would argue if we are using these for testing purposes we should just have it all in one place. Open to hearing counter args but bouncing between two notebooks is a bit annoying for testing and I always end up just merging the notebooks.

File: knoten/vis.py [hunk 2]
```diff
@@ -133,10 +133,174 @@ def point_info(cube_path, x, y, point_type, allow_outside=False):
...
 
-def reprojection_diff(isd, cube, nx=4, ny=8):
+def plot_diff(data, title='diff plot', colx='x', coly='y', coldx='diffx', coldy='diffy', colmag='magnitude', width=500, height=500):
+    import matplotlib.cm as cm
+    from matplotlib.colors import Normalize
+
+    fig = make_subplots(rows=2, cols=2, column_widths=[0.9, .1], row_width=[.9, .1],
+                        shared_xaxes=True, shared_yaxes=True, horizontal_spacing = 0.01, vertical_spacing = 0.01)
+
+    arrow_scale = (data[coly].iloc[1]-data[colx].iloc[0]+data[colx].iloc[1]-data[colx].iloc[0])/5
```
Comment (jessemapel) [inline knoten/vis.py]: should this be `coly[1] - coly[0] + colx[1] - colx[0]`?

Comment (jessemapel) [reply knoten/vis.py]: Is this variable even used? scale looks hard coded to 1

**Comment by krlberry** [comment]
> I would argue if we are using these for testing purposes we should just have it all in one place. Open to hearing counter args but I bouncing between two notebooks is a bit annoying for testing and I end up just merging the notebooks.

I agree. I also ended up merging the notebooks for testing so that ISD generation was part of the process.

**Comment by Kelvinrr** [comment]
I'm updating the notebook one more time,  so the results are from the latest ALE version. Then this should be good to go.

File: knoten/vis.py (hunk 2 above)
Comment (Kelvinrr) [reply knoten/vis.py]: It's no longer used, it was before I decided to leave the quiver scale to one so the length is one to one with the pixel offset.

**Review by jessemapel — APPROVED** [review]

**Pull request merged**